### ISSUE TITLE
ValueError: [digital envelope routines: EVP_DigestInit_ex] disabled for FIPS In Oracle GoldenGate 21c microservices on Linux

### ISSUE THREAD

**Issue by abhijitvk** [oracle/docker-images#2394]
Error Stacktrace from Oracle GoldenGate 21c microservice container start-up for out-of-the-box deployment as below. 

The build and docker run steps are all documented in the attached file for reference.

`Traceback (most recent call last):
  File "/usr/local/bin/deployment-init.py", line 265, in <module>
    sys.exit(main())
  File "/usr/local/bin/deployment-init.py", line 260, in main
    create_ogg_deployment()
  File "/usr/local/bin/deployment-init.py", line 251, in create_ogg_deployment
    reset_service_configuration(serviceName)
  File "/usr/local/bin/deployment-init.py", line 166, in reset_service_configuration
    config = get_service_config(serviceName)
  File "/usr/local/bin/deployment-init.py", line 140, in get_service_config
    response = get_requests_session().get(url, headers=rest_call_headers, auth=get_digest_authentication())
  File "/usr/lib/python3.9/site-packages/requests/sessions.py", line 555, in get
    return self.request('GET', url, **kwargs)
  File "/usr/lib/python3.9/site-packages/requests/sessions.py", line 542, in request
    resp = self.send(prep, **send_kwargs)
  File "/usr/lib/python3.9/site-packages/requests/sessions.py", line 662, in send
    r = dispatch_hook('response', hooks, r, **kwargs)
  File "/usr/lib/python3.9/site-packages/requests/hooks.py", line 31, in dispatch_hook
    _hook_data = hook(hook_data, **kwargs)
  File "/usr/lib/python3.9/site-packages/requests/auth.py", line 267, in handle_401
    prep.headers['Authorization'] = self.build_digest_header(
  File "/usr/lib/python3.9/site-packages/requests/auth.py", line 185, in build_digest_header
    HA1 = hash_utf8(A1)
  File "/usr/lib/python3.9/site-packages/requests/auth.py", line 148, in md5_utf8
    return hashlib.md5(x).hexdigest()
ValueError: [digital envelope routines: EVP_DigestInit_ex] disabled for FIPS`

[github-issue-capture-steps.txt](https://github.com/oracle/docker-images/files/8909774/github-issue-capture-steps.txt)

**Comment by sbalousek** [comment]
OK. Maybe not. My test with 'Digest-SHA-256' shows:

```
...
2022-06-15T15:42:25.090Z | INFO    | Step 3: Using an existing ServiceManager (127.0.0.1:9011).
2022-06-15T15:42:25.090Z | INFO    | 
2022-06-15T15:42:25.090Z | INFO    | Step 4: Create Deployment.
2022-06-15T15:42:25.090Z | INFO    | Connecting to ServiceManager at http://127.0.0.1:9011 to issue a POST request for '/services/v2/deployments/Local'
2022-06-15T15:42:25.090Z | INFO    | Auth user: 'oggadmin'
2022-06-15T15:42:25.140Z | SEVERE  | > POST http://127.0.0.1:9011/services/v2/deployments/Local HTTP/1.1
2022-06-15T15:42:25.140Z | SEVERE  | > Accept-Charset: UTF-8
2022-06-15T15:42:25.140Z | SEVERE  | > Accept: application/json
2022-06-15T15:42:25.140Z | SEVERE  | > Content-Type: application/json
2022-06-15T15:42:25.140Z | SEVERE  | >
2022-06-15T15:42:25.140Z | SEVERE  | {"passwordRegex":"(?=^.{8,30}$)(?=(.*[0-9]))(?=(.*[A-Z]))(?=(.*[a-z]))(?=(.*[-!@%&*.#_]))(?!(.*[$^?]))^.*","$schema":"ogg:deployment","oggConfHome":"/u02/Deployment/etc/conf","oggVarHome":"/u02/Deployment/var","oggHome":"/u01/ogg","oggEtcHome":"/u02/Deployment/etc","oggSslHome":"/u02/Deployment/etc/ssl","enabled":true,"oggDataHome":"/u02/Deployment/var/lib/data","status":"running"}< HTTP/1.1 401 Unauthorized
2022-06-15T15:42:25.141Z | SEVERE  | < Cache-Control: max-age=0, no-cache, no-store, must-revalidate
2022-06-15T15:42:25.141Z | SEVERE  | < X-Content-Type-Options: nosniff
2022-06-15T15:42:25.141Z | SEVERE  | < Content-Security-Policy: default-src 'self' 'unsafe-eval' 'unsafe-inline';img-src 'self' data:;frame-ancestors http://127.0.0.1:9011;child-src http://127.0.0.1:9011 blob:;
2022-06-15T15:42:25.141Z | SEVERE  | < WWW-Authenticate: [Basic realm="ServiceManager",charset="UTF-8", Digest realm="ServiceManager",algorithm=sha-256,qop="auth",nonce="11789266786025026812"]
2022-06-15T15:42:25.141Z | SEVERE  | < Pragma: no-cache
2022-06-15T15:42:25.141Z | SEVERE  | < Expires: 0
2022-06-15T15:42:25.141Z | SEVERE  | < X-XSS-Protection: 1; mode=block
2022-06-15T15:42:25.141Z | SEVERE  | < Content-Length: 321
2022-06-15T15:42:25.141Z | SEVERE  | < Content-Type: application/json
2022-06-15T15:42:25.141Z | SEVERE  | <
2022-06-15T15:42:25.141Z | SEVERE  | Error code: 401
2022-06-15T15:42:25.141Z | SEVERE  | Error(s) ocurred, please check log files.
2022-06-15T15:42:25.141Z | SEVERE  | Failed to register deployment. Make sure Service Manager is running and the administrator user/password specified is correct. Check the Configuration Assistant and Service Manager log files for additional details.
2022-06-15T15:42:25.141Z | SEVERE  | Deployment (Local) creation failed. Check log file(s).
```
It looks like the OGG installer does not support the SHA-156 mode of Digest Authentication.

**Comment by sbalousek** [comment]
I'm also interested in the version of Docker you're running. Would you be able top provide the output from `docker info`, as well?

**Comment by sbalousek** [comment]
Although this seems like an environment issue to me, the correct approach of using `Digest-SHA-256` is not going to work because the GoldenGate installation framework does not support that algorithm for Digest Authentication.

Before falling back to Basic Authentication, is it possible that you can try changing the two instances of 'Digest,Basic' to 'Digest-SHA-256,Digest,Basic' in `bin/deployment-init.py`?  That would make the default Digest Authentication algorithm be SHA-256 (for Python), but still allow the authentication using the deprecated MD5 algorithm (for the OGG installer).

**Comment by abhijitvk** [comment]
Great suggestion @sbalousek - That change actually did work, so I went ahead and updated the deployment-init.py in the PR itself. Thanks for your follow-up

**Comment by sbalousek** [comment]
Thank you for working through this issue. I've already contacted the OGG installer team requesting support for the SHA-256 algorithm for Digest Authentication.

**Comment by sbalousek** [comment]
Resolved with PR #2385 and #2396
